src/http/data_loader.rs

```diff
@@ -5,31 +5,48 @@ use std::time::Duration;
 
 use anyhow::Result;
 use async_graphql::async_trait;
-use async_graphql::dataloader::{DataLoader, HashMapCache, Loader};
+use async_graphql::dataloader::{DataLoader, HashMapCache, Loader, NoCache};
 use async_graphql::futures_util::future::join_all;
 use async_graphql_value::ConstValue;
-use reqwest::header::HeaderMap;
 use url::Url;
 
 use crate::http::{HttpClient, Method, Response};
 use crate::json::JsonLike;
-#[derive(Clone, Debug, Eq, PartialEq)]
+#[derive(Debug)]
 pub struct EndpointKey {
-  pub url: Url,
-  pub headers: HeaderMap,
-  pub method: Method,
+  pub request: reqwest::Request,
   pub match_key_value: ConstValue,
   pub match_path: Vec<String>,
   pub batching_enabled: bool,
   pub list: bool,
 }
-
 impl Hash for EndpointKey {
   fn hash<H: Hasher>(&self, state: &mut H) {
-    self.url.hash(state);
-    self.match_key_value.to_string().hash(state);
+    self.request.url().hash(state);
+    self.request.method().hash(state);
+    // self.request.headers().hash(state);
+  }
+}
+
+impl PartialEq for EndpointKey {
+  fn eq(&self, other: &Self) -> bool {
+    self.request.url() == other.request.url() && self.request.method() == other.request.method()
+  }
+}
+
+impl Clone for EndpointKey {
+  fn clone(&self) -> Self {
+    EndpointKey {
+      request: self.request.try_clone().unwrap(),
```

Comment:
This can fail. Ideally EndpointKey is only applicable for GET requests. We don't need the complete request here to clone.